### PR TITLE
ddl: forbid truncating sequence object

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Problem Summary: forbid truncating sequence object.

### What is changed and how it works?

What's Changed: forbid truncating sequence object with `truncate table seq`

How it Works:
judge the truncated object in DDL worker

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
- ddl: forbid truncating sequence object

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3753,6 +3753,9 @@ func (d *ddl) TruncateTable(ctx sessionctx.Context, ti ast.Ident) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if tb.Meta().IsView() || tb.Meta().IsSequence() {
+		return infoschema.ErrTableNotExists.GenWithStackByArgs(schema.Name.O, tb.Meta().Name.O)
```
Comment (djshow832) [inline ddl/ddl_api.go:3757]: `ErrTableNotExists` is not suitable.

Comment (AilinKid) [reply ddl/ddl_api.go:3757]: MySQL throws this error when truncating view.
MariaDB throws this error when truncating these two.

**Comment by AilinKid** [comment]
> Do we need to forbid “lock table”？

good, addressed.

**Pull request merged**